conjure-java-core/src/main/java/com/palantir/conjure/java/types/UnionGenerator.java

```diff
@@ -190,30 +201,310 @@ private static MethodSpec generateEquals(ClassName unionClass) {
                         .returns(TYPE_VARIABLE)
                         .build())
+                .addMethod(MethodSpec.methodBuilder("builder")
+                        .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+                        .addTypeVariable(TYPE_VARIABLE)
+                        .addStatement("return new $T<$T>()",
+                                visitorBuilderClass,
+                                TYPE_VARIABLE)
+                        .returns(ParameterizedTypeName.get(
+                                visitorStageInterfaceName(unionClass,
+                                memberTypes.keySet().iterator().next().get()),
+                                TYPE_VARIABLE))
+                        .build())
                 .build();
     }
 
     private static List<MethodSpec> generateMemberVisitMethods(Map<FieldName, TypeName> memberTypes) {
         return memberTypes.entrySet().stream().map(entry -> {
             String variableName = variableName();
-            return MethodSpec.methodBuilder(VISIT_METHOD_NAME + StringUtils.capitalize(entry.getKey().get()))
+            return MethodSpec.methodBuilder(visitMethodName(entry.getKey().get()))
                     .addModifiers(Modifier.PUBLIC, Modifier.ABSTRACT)
                     .addParameter(entry.getValue(), variableName)
                     .returns(TYPE_VARIABLE)
                     .build();
         }).collect(Collectors.toList());
     }
 
+    /**
+     * Generates the Visitor builder class.
+     *
+     * @param enclosingClass the enclosing class
+     * @param visitor the interface of the Visitor
+     * @param visitorBuilder the ClassName of the class to generate
+     * @param memberTypeMap mapping from member names to member types
+     * @return
+     */
+    private static TypeSpec generateVisitorBuilder(
+            ClassName enclosingClass,
+            ClassName visitor,
+            ClassName visitorBuilder,
+            Map<FieldName, TypeName> memberTypeMap) {
+        TypeVariableName visitResultType = TypeVariableName.get("T");
+        return TypeSpec.classBuilder(visitorBuilder)
+                .addModifiers(Modifier.PRIVATE, Modifier.STATIC)
+                .addTypeVariable(visitResultType)
+                .addSuperinterfaces(allVisitorBuilderStages(enclosingClass, memberTypeMap.keySet(), visitResultType))
+                .addFields(allVisitorBuilderFields(memberTypeMap, visitResultType))
+                .addMethods(allVisitorBuilderSetters(enclosingClass, visitResultType, memberTypeMap))
+                .addMethod(builderBuildMethod(visitor, visitResultType, memberTypeMap))
+                .build();
+    }
+
+    /**
+     * Generates setter methods for the Visitor builder on the following format.
+     *
+     * NextMemberStage&lt;T&gt; memberName(Function&lt;MemberType, T&gt; memberVisitor) {
+     *     this.memberVisitor = memberVisitor;
+     *     return this;
+     * }
+     *
+     * @param unionClass The enclosing class
```

Comment:
unless this has changed recently, the code style in this repo favours explanation of parameters as part of the text. IMO, it's also fine to keep these pretty short in single-use internal/private methods.

Reply:
Good points. PR updated accordingly.

(1+3) Right, the interface is probably good to keep. It allows handler classes to do things like `something.accept(this)` and it probably makes it slightly nicer to implement more complex stateful handlers.

(2) A sample stack trace...

    Exception in thread "main" java.lang.RuntimeException: test
        at conjuretesting.Main.lambda$main$1(Main.java:34)
        at org.aioobe.TestAnimal$VisitorBuilder$1.visitCat(TestAnimal.java:126)
        at org.aioobe.TestAnimal.accept(TestAnimal.java:45)
        at conjuretesting.Main.main(Main.java:32)

...where the top stack entry points at the throw statement in the lambda expression. So quite good I'd say.

Happy to do another iteration if anything else comes up. Otherwise, what would the next step be? How do I deal with the changelog check?

Reply:
Either add a changelog entry manually or click the `Generate changelog entry` box towards the top of the comment section.

Reply:
I'm happy with this PR, but let's give @iamdanfox a chance to take a look.